BridgeCareApp/BridgeCareCodeFirst/Models/BudgetReportData.cs

```diff
@@ -0,0 +1,137 @@
+            var db = new BridgeCareContext();
+            string getReport =
+                "SELECT Years, Budget, Cost_ " +
+                    " FROM Report_" + data.NetworkId
+                    + "_" + data.SimulationId + " WHERE BUDGET is not null";
+            try
+            {
+                var RawQueryForData = db.Database.SqlQuery<BudgetReport>(getReport).AsQueryable();
+
+                double sum = 0;
+                foreach (var report in RawQueryForData)
+                {
+                    double.TryParse(report.Cost_.Value.ToString(), out double cost);
+                    Hashtable yearView;
+                    if (!BudgetYearView.Contains(report.Budget))
+                    {
+                        yearView = new Hashtable();
+                        BudgetYearView.Add(report.Budget, yearView);
+                    }
+                    else
+                    {
+                        yearView = (Hashtable)BudgetYearView[report.Budget];
+                    }
+
+                    if (yearView.Contains(report.Years))
+                    {
+                        sum = (double)yearView[report.Years];
+                        yearView.Remove(report.Years);
+                    }
+                    sum += cost;
+                    yearView.Add(report.Years, sum);
+                    costList.Add(new CostDetails
+                    {
+                        Cost = report.Cost_.Value,
+                        Years = report.Years,
+                        Budget = report.Budget
+                    });
+                }
+                foreach (string item in BudgetYearView.Keys)
+                {
+                    if (!budgetTypes.Contains(item))
+                    {
+                        BudgetYearView.Remove(item);
+                    }
+                }
+            }
+            catch (SqlException ex)
+            {
+                db.Dispose();
+                if (ex.Number == -2 || ex.Number == 11)
+                {
+                    throw new TimeoutException("The server has timed out. Please try after some time");
+                }
+                if (ex.Number == 208)
+                {
+                    throw new InvalidOperationException("Network or simulation table does not exist in the database");
+                }
+            }
+            catch (OutOfMemoryException)
+            {
+                db.Dispose();
+                throw new OutOfMemoryException("The server is out of memory. Please try after some time");
+            }
+            db.Dispose();
+            var budgetReportDetails = new BudgetReportDetails
+            {
+                BudgetForYear = BudgetYearView,
+                CostDetails = costList
+            };
+            return budgetReportDetails;
+        }
+
+        public string[] InvestmentData(SimulationResult data)
+        {
+            var db = new BridgeCareContext();
```

Comment:
If this stays in, which it may not - consider using "using" statements instead of always having to call .Dispose()

Reply:
Removed Dispose. Entity framework handles this. If in some case it does not. We can handle this manually.